presto-pinot/src/main/java/io/prestosql/pinot/query/DynamicTableBuilder.java

```diff
@@ -122,6 +125,32 @@ public static DynamicTable buildFromPql(PinotMetadata pinotMetadata, SchemaTable
         return new DynamicTable(pinotTableName, suffix, selectionColumns, groupByColumns, filter, aggregationExpressionBuilder.build(), orderBy, getTopNOrLimit(request), getOffset(request), query);
     }
 
+    public static Optional<String> getAggregationFunctionName(AggregateFunction aggregateFunction)
+    {
+        requireNonNull(aggregateFunction, "aggregateFunction is null");
+        if (aggregateFunction.isDistinct()) {
+            if (aggregateFunction.getFunctionName().equalsIgnoreCase("count")) {
+                // When aggregation expressions push down is implemented,
+                // count(distinct <expression>) will be mapped to distinctcount(expression)
+                // See https://github.com/prestosql/presto/issues/4171
+                // Until then this branch will never be executed
+                return Optional.of("distinctcount");
+            }
+            return Optional.empty();
+        }
+        return Optional.of(aggregateFunction.getFunctionName());
+    }
+
+    public static String getInputExpresion(AggregateFunction aggregateFunction)
+    {
+        requireNonNull(aggregateFunction, "aggregateFunction is null");
+        // Pinot converts count(column_name) to count(*)
+        if (aggregateFunction.getInputs().isEmpty() || getAggregationFunctionName(aggregateFunction).map(functionName -> functionName.equalsIgnoreCase("count")).orElse(false)) {
+            return "*";
```

Comment:
This is hard to read, especially with the optional and lambda in the middle. I'd break it up into multiple parts and make the check more explicit.